andela/wg-rogue-one
# #157731626 Add support for macrocycle, mesocycle and microcycle

### What does this PR do?
Adds support for macrocycle, mesocycle and microcycle
#### Description of Task to be completed?
1. Run Autopep8 on the code base
#### How should this be manually tested?
1. Run the commands below
```
$ git clone https://github.com/andela/wg-rogue-one.git

$ git checkout ft-support-macrocycle-mesocycle-157731626

```
follow the setup instructions in [README.rst](https://github.com/andela/wg-rogue-one/blob/ft-support-macrocycle-mesocycle-157731626/README.rst)
2. Run the app with `python manage.py runserver`
3. Login to wger 
4. Navigate to the `Workout overview` and add a new workout. 
5. You should be able to choose a training cycle. 
<img width="759" alt="screen shot 2018-06-22 at 12 43 49" src="https://user-images.githubusercontent.com/13131669/41770150-1a7e973c-761a-11e8-9114-1e04112c9037.png">

6. Choose between `Microcycle`, `Mesocycle`and `Macrocycle`.
7. <img width="1437" alt="screen shot 2018-06-22 at 11 59 53" src="https://user-images.githubusercontent.com/13131669/41769986-9a8535cc-7619-11e8-86c8-ab92f626d80a.png">
Note: Added workouts should have a cycle indicator


#### Any background context you want to provide?
Macro is an annual plan, meso is phase of training with a duration of between 2-6 weeks and micro is instead only a week.

This is needed for sports periodization ( powerlifting, weightlifting, strongman, ecc ) or at least to create a fitness program drafted on more than only one week.

#### What are the relevant pivotal tracker stories?
[#157731626](https://www.pivotaltracker.com/story/show/157731626)
